src/main/java/org/mule/extension/sftp/internal/connection/SftpClient.java

```diff
@@ -376,12 +389,29 @@ public void write(String path, InputStream stream, FileWriteMode mode) throws IO
    */
   public String getHome() {
     try {
-      return session.executeRemoteCommand("pwd").trim();
+      return executePWDCommandWithTimeout();
     } catch (IOException e) {
       throw new RuntimeException(e);
     }
   }
 
+  /**
+   * Apache Mina 2.9.2 - have a hardcoded timeout to infinite on executeRemoteCommand, if this method fails, the operation hangs.
+   * The solution at the moment is to add a timeout with the tools provided by mule sdk.
+   */
+  public String executePWDCommandWithTimeout() throws IOException {
```

Comment:
Shouldn't it be enough that the exception is IllegalPathException? Why are you going for a more generic one?